composeApp/src/main/kotlin/io/github/couchtracker/ui/components/CastPortrait.kt

```diff
@@ -0,0 +1,99 @@
+import app.moviebase.tmdb.image.TmdbImageType
+import app.moviebase.tmdb.model.TmdbCast
+import coil3.compose.AsyncImage
+import io.github.couchtracker.tmdb.TmdbPerson
+import io.github.couchtracker.tmdb.TmdbPersonId
+import io.github.couchtracker.ui.ImageModel
+import io.github.couchtracker.ui.ImagePreloadOptions
+import io.github.couchtracker.ui.PlaceholdersDefaults
+import io.github.couchtracker.ui.rememberPlaceholderPainter
+import io.github.couchtracker.ui.toImageModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+
+@Composable
+fun CastPortrait(
+    modifier: Modifier,
+    person: CastPortraitModel?,
+    onClick: ((CastPortraitModel) -> Unit)?,
+) {
+    PortraitComposable(
+        modifier,
+        image = { w, h ->
+            if (person != null) {
+                AsyncImage(
+                    model = person.posterModel?.getCoilModel(w, h),
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .clickable(onClick != null) { onClick?.invoke(person) },
+                    contentDescription = null,
+                    contentScale = ContentScale.Crop,
+                    fallback = rememberPlaceholderPainter(PlaceholdersDefaults.PERSON.icon, isError = false),
+                    error = rememberPlaceholderPainter(PlaceholdersDefaults.PERSON.icon, isError = true),
+                )
+            }
+        },
+        label = {
+            Text(
+                text = person?.name.orEmpty(),
+                textAlign = TextAlign.Center,
+                minLines = 1,
+            )
+            Text(
+                text = person?.character.orEmpty(),
+                textAlign = TextAlign.Center,
+                minLines = 1,
+                // Same style as ListItem's supporting content
+                color = MaterialTheme.colorScheme.onSurfaceVariant,
+                style = MaterialTheme.typography.bodyMedium,
+            )
+        },
+    )
+}
+
+data class CastPortraitModel(
+    val person: TmdbPerson,
+    val name: String,
+    val posterModel: ImageModel?,
+    val character: String,
+) {
+    companion object {
+
+        suspend fun fromTmdbCast(
+            cast: TmdbCast,
+            imagePreloadOptions: ImagePreloadOptions,
+        ): CastPortraitModel {
+            return CastPortraitModel(
+                person = TmdbPerson(TmdbPersonId(cast.id)),
+                name = cast.name,
+                character = cast.character,
+                posterModel = cast.profilePath?.let { path ->
+                    TmdbImage(path, TmdbImageType.PROFILE).toImageModel(imagePreloadOptions)
+                },
+            )
+        }
+    }
+}
+
+suspend fun List<TmdbCast>.toCastPortraitModel(
+    imagePreloadOptions: ImagePreloadOptions,
+): List<CastPortraitModel> = coroutineScope {
+    map {
+        async(Dispatchers.IO) {
```

Comment:
Why are changing dispacher here? `CastPortraitModel.fromTmdbCast` is suspend, so shouldn't we set `Dispatchers.IO` closer to where the blocking call is? Also, is there even blocking call here?